### ISSUE TITLE
[Feature Request] Compare conflicted files to their ancestor

### ISSUE THREAD

**Issue by Tricertops** [ForkIssues/Tracker#1325]
When viewing a diff of two files or resolving a merge, it would be very helpful to not just see the difference between the two conflicted files, but also the difference to their common ancestor. This would especially be useful for deletions, since it would let you see _what_ was deleted.

Consider this example:

Common ancestor:
```
A
B
C
D
```

Branch 1, deleted B and C:
```
A
D
```

Branch 2, added a New line:
```
A
B
C
New
D
```

The diff shows this:
```
A          A
           B
           C
           New
D          D
```

It’s not clear from this view that B and C were deleted <sub>(note that typical lines of code don’t follow alphabet 😄)</sub>. The New line is technically added _after_ the deleted lines, so this is not even a real conflict. I think it’s too much to ask for an improved diff algorithm, so maybe just an ability to see the original file in this the diff view would be enough. Maybe the best would be to show the deletion in-line, so the user can cherry-pick the actual addition:

```
A          A
-B-        B
-C-        C
           New
D          D
```

**Comment by blindjimmy** [comment]
Yes, this is an extremely useful feature when dealing with repos that have many people working on the same files.

However, I will point out that the way some merge tools solve this (p4merge and KDiff3 being good examples) is not by updating their diff algorithm, but instead by adding a 4th read-only pane to the merge UI showing the common ancestor. So in the example above what we'd see when resolving the conflict is:

**Common Ancestor Pane**
```
A
B
C
D
```

**Branch 1 Pane**
```
A
D
```

**Branch 2 Pane**
```
A
B
C
New
D
```

**Resolution Pane**
```
A
<conflict to be resolved>
D
```

So simply by being able to see what the lines were like before either branch changed them, it is now clear that lines B and C were deleted by Branch 1.

https://stackoverflow.com/questions/6022365/3-way-merge-tools-for-mac-that-show-4-panes and https://stackoverflow.com/questions/2621475/three-way-merge-different-philosophies both talk about this strategy some.

So please, please put me down as a heavy +1 for Fork supporting at least an optional 4th pane in its merge UI for the common ancestor... this is the only thing that keeps me from using Fork for all my merges; I currently have to fall back to p4merge when things get confusing. :)
